### DISCUSSION TITLE
Override base styles scoped to a container, yet still allowing for element-level customization

### DISCUSSION THREAD

**Discussion by mcmire** [tailwindlabs/tailwindcss#2452] [Help] [answered]
This might be my inexperience with using a utility-first framework showing, but I have a problem where I want to override the styles for, say, links across the board, but because I don't want these styles to be truly global (say, there is a sidebar that also contains navigation links and I want there to be a clean slate there), I want to scope them to a container. For instance:

``` css
.main-content a {
  @apply text-blue-500;
}

.main-content a:hover {
  @apply border-b border-blue-500;
}

.main-content a:active, .container a:visited {
  @apply text-gray-500 border-gray-500;
}
```

So far, so good. Now I want to override these styles on a per-element basis, so, for instance:

``` html
<div class="container">
  <div class="sidebar">
    <!-- ... -->
  </div>
  <div class="main-content">
    <!-- ... some kind of form ... -->
    <a href="#" class="hover:border">Cancel</a>
  </div>
</div>
```

However, this doesn't work, because due to specificity, `.main-content a` will always win.

If I weren't using Tailwind, I think the way I would fix this is to match the same specificity when applying styles to the link, so:

``` html
<div class="container">
  <div class="sidebar">
    <!-- ... -->
  </div>
  <div class="main-content">
    <!-- ... some kind of form ... -->
    <a href="#" class="action-link">Cancel</a>
  </div>
</div>
```

``` css
.main-content .action-link:hover {
  border-width: 1px;
}
```

What is the Tailwind-idiomatic way to handle this kind of thing? Is overriding styles globally like this a good idea, or is it an anti-pattern? Is it preferred to always specify classes you want every time, even if it's for an element you know is going to be fairly ubiquitous?

**Comment by samzlab** [comment, marked as answer]
If you enable the `important` option in your `tailwind.config.js` then the direct `hover:border` classes will have the "!important" suffixes for each rule. If I remember well the `important` option does not affect the styles generated by the @apply function so its should work (just a tip, did not tried it out):

```
// tailwind.config.js
module.exports = {
  important: true,
}
```
See: https://tailwindcss.com/docs/configuration#important

**Reply by mcmire** [reply]
Coming back to this again because I thought I figured out a way around this but apparently I didn't! I'm still wondering if there is a less nuclear option, but it does work, so thanks for this.
